default.nix

```diff
@@ -113,29 +113,7 @@ let
         # Provide a dummy installPhase for benchmark packages.
         installPhase = "mkdir -p $out";
       }) // optionalAttrs (args ? src) {
-        src = let
-           cleanSourceFilter = with pkgs.stdenv;
-             name: type: let baseName = baseNameOf (toString name); in ! (
-               # Filter out .git repo
-               (type == "directory" && baseName == ".git") ||
-               # Filter out editor backup / swap files.
-               lib.hasSuffix "~" baseName ||
-               builtins.match "^\\.sw[a-z]$" baseName != null ||
-               builtins.match "^\\..*\\.sw[a-z]$" baseName != null ||
-
-               # Filter out locally generated/downloaded things.
-               baseName == "dist" ||
-
-               # Filter out the files which I'm editing often.
-               lib.hasSuffix ".nix" baseName ||
-               # Filter out nix-build result symlinks
-               (type == "symlink" && lib.hasPrefix "result" baseName)
-             );
-
-          in
-            if (builtins.typeOf args.src) == "path"
-              then builtins.filterSource cleanSourceFilter args.src
-              else args.src or null;
+        src = localLib.filterSource args.src;
```

Comment:
As `filterSource` is implemented I think some filters are getting dropped, like:

* `# Filter out .git repo`
* `# Filter out editor backup / swap files.`
* `# Filter out nix-build result symlinks`

Is that what we want?

Reply:
The good thing about `pkgs.lib.cleanSourceWith` (as opposed to `builtins.filterSource`) is that it's chainable. In `localLib.filterSource` it calls the "default" `pkgs.lib.cleanSource` function which provides the above filters. I verify this by running nix-build and checking that a cached build is used.